usaspending_api/etl/es_etl_helpers.py

```diff
@@ -245,13 +247,24 @@ def convert_postgres_array_as_string_to_list(array_as_string: str) -> list:
     return array_as_string[1:-1].split(",") if len(array_as_string) > 2 else None
 
 
-def convert_postgres_json_array_as_string_to_json(json_array_as_string: str) -> dict:
+def convert_postgres_json_array_as_string_to_list(json_array_as_string: str) -> Optional[dict]:
     """
-        Postgres JSON arrays (jsonb) are stored in CSVs as strings. Elasticsearch is able to handle
-        JSON arrays with nested types, but needs to be passed a list JSON instead of a string.
-        In the case of an empty array, return null.
+        Postgres JSON arrays (jsonb) are stored in CSVs as strings. Since we want to avoid nested types
+        in Elasticsearch the JSON arrays are converted to dictionaries to make parsing easier and then
+        converted back into a formatted string.
     """
-    return json.loads(json_array_as_string) if json_array_as_string else None
+    if json_array_as_string is None or len(json_array_as_string) == 0:
+        return None
+    result = []
+    json_array = json.loads(json_array_as_string)
```

Comment:
Is this guaranteed to create a list preserving order? Such that the components will always be pulled from the DB and pushed to ES always in the same order to satisfy  the search?